### DISCUSSION TITLE
How to Change wind profile parameters

### DISCUSSION THREAD

**Discussion by peterfakhry** [OpenFAST/openfast#1117] [Q&A]
Hello there
I have a numerical model following the Staino et al. 2013 (for solving the edgewise and flapwaise blade displacements) and i am in the process of validating this model using openfast. I simulate the turbulence found in the wind using the following equations
![image](https://user-images.githubusercontent.com/40067011/167136125-03d9de01-6b38-4dde-b001-e6024f1b23e5.png)

 In the openfast I don't know how to change parameters like the roughness length, von karman and the turbulence intensity. I can see from the `NRELOffshrBsline5MW_InflowWind_12mps` file that the default windtype is 3 and to change their parameters is through `90m_12mps_twr.bts` which is not a readable file. My question is : what is the appropriate windtype that matches my case and how to change the turbulence parameters
Thanks a lot

**Comment by jjonkman** [comment]
Dear @peterfakhry,

Full field turbulent wind data is an input to OpenFAST.  You can use TurbSim in a preprocessing step to generate this full field turbulent wind data.  If one of the existing turbulence spectra models supported by TurbSim (e.g., Kaimal or van Karman) does not meet your needs, you can always implement your own turbulence spectra by setting `TurbModel` = "USRINP".  See the TurbSim documentation for more information: https://www.nrel.gov/wind/nwtc/assets/downloads/TurbSim/TurbSim_v2.00.pdf.

Best regards,

**Reply by peterfakhry** [reply]
Thank you so much for your answer Prof. Jonkman. 
sorry but i am a beginner. I shall read the attached documentation about TurbSim but it I don't have an executable for it (is it embedded in OpenFAST?). What I do in my model is that I generate the Kaimal spectra (using the above equations) and use iFFT to get the time domain values for turbulence and apply it in the X direction. I think, in OpenFAST, this corresponds to WindType 5 but when I try to use it I get the error that i dont have these files in my directory of the basline folder `"wasp\Output\basic_5u.bin"`, `"wasp\Output\basic_5v.bin"` and `"wasp\Output\basic_5w.bin"`. 
Thank you

**Reply by jjonkman** [reply]
Dear @peterfakhry,

TurbSim is not embedded in OpenFAST.  Instead, TurbSim is run separately in a pre-processing step.  A precompiled Windows executable of TurbSim is available here: https://github.com/OpenFAST/openfast/releases/download/v3.1.0/TurbSim_x64.exe.  (Apart from Windows, you can compile TurbSim using the source code provided within OpenFAST.)

Does you approach consider the spatial variability of turbulence across the disk based on spatial coherence functions or is the turbulence uniform across the disk?  If the latter, presumably you'd generate a uniform wind data file (`WindType` = 2 in InflowWind).  If spatial variability is accounted for, what format are you writing the wind data in?

TurbSim considers both the turbulent spectra and spatial variability and can generate wind data files supported by `WindType` =3 and 4 in InflowWind.

Best regards,

**Reply by peterfakhry** [reply]
Thank you for your answer
No sir my approach doesn't consider the spatial variability of turbulence across the disk. Isotropic one dimensional homogenous uniform turbulence has been assumed over the rotor field. I changed the `WindType` to 2 but i got the following error
```
FAST_InitializeAll:InflowWind_Init:Uniform_ValidateInput: Cannot find uniform wind input file:
'.\../5MW_Baseline/unused'

 OpenFAST encountered an error during module initialization.
 Simulation error level: FATAL ERROR

 Aborting OpenFAST.
```
Another question, how do we change the mean wind speed? Please excuse me if my questions seem trivial. 
Thanks a lot

**Reply by jjonkman** [reply]
Dear @peterfakhry,

To use `WindType` = 2, you must write your uniform wind data into a file with the proper format.  See the InflowWind User's Guide for more information: https://www.nrel.gov/wind/nwtc/assets/downloads/InflowWind/InflowWind_Manual.pdf.  This wind file will include the mean wind speed.

Best regards,
